pyvista/core/dataset.py

```diff
@@ -2523,3 +2523,54 @@ def cell_type(self, ind: int) -> int:
 
         """
         return self.GetCellType(ind)
+
+    def point_is_inside_cell(
+        self, ind: int, point: Union[VectorArray, NumericArray]
+    ) -> Union[int, np.ndarray]:
+        """Return whether a point is inside a cell.
+
+        Parameters
+        ----------
+        ind : int
+            Cell ID.
+
+        point : Sequence(float) or np.ndarray
+            Coordinates of point to query (length 3) or a ``numpy`` array of ``n``
+            points with shape ``(n, 3)``.
+
+        Returns
+        -------
+        bool or numpy.ndarray
+            Whether point(s) is/are inside cell.
+
+        Examples
+        --------
+        >>> from pyvista import examples
+        >>> mesh = examples.load_hexbeam()
+        >>> mesh.cell_bounds(0)
+        (0.0, 0.5, 0.0, 0.5, 0.0, 0.5)
+        >>> mesh.point_is_inside_cell(0, [0.2, 0.2, 0.2])
+        True
+
+        """
+        if not 0 <= ind < self.n_cells:
+            raise ValueError(f"ind must be >= 0 and < {self.n_cells}, got {ind}")
+
+        point = _coerce_pointslike_arg(point, copy=False)
+
+        cell = self.GetCell(ind)
+        npoints = cell.GetPoints().GetNumberOfPoints()
+
+        closest_point = [0.0, 0.0, 0.0]
+        sub_id = _vtk.mutable(0)
+        pcoords = [0.0, 0.0, 0.0]
+        dist2 = _vtk.mutable(0.0)
+        weights = [0.0] * npoints
+
+        in_cell = [
+            bool(cell.EvaluatePosition(node, closest_point, sub_id, pcoords, dist2, weights))
+            for node in point
+        ]
+        if len(in_cell) == 1:
+            return in_cell[0]
```

Comment:
In the edge case of someone passing in a `(1, 3)`-shaped arraylike, is it OK that this will return `True` instead of `[True]`? If the API says "return a scalar for a scalar", it might be more consistent to respect if the input is `[(0, 0, 0)]` instead of `(0, 0, 0)`. But unpacking the singular case is fine too, just something to consider.

Reply:
Good point.  We currently have this problem in the other functions too: 

https://github.com/pyvista/pyvista/blob/f8208ad25c518d3338e57a821a9bd0772286b08b/pyvista/core/dataset.py#L2285

Reply:
I think this is important to fix.  A user might be dynamically generating a list of points to use, it will be unexpected that they should have to check for arrays of shape (1, 3).

Since _points_like_arg is a private method, we have free reign to change it.  I'm not sure if we should take extra steps to warn users that the behavior has changed, or not.  I argue that the current behavior is a bug, so we can just change it.

Reply:
Although fixing the bug mostly means that some ifs in downstream code will merely be redundant, we should probably annotate the involved public functions with `.. versionadded:`. Otherwise people may never notice.

Reply:
If you meant emitting warnings: it could go either way. Not strictly necessary, but this is a very high signal to noise ratio warning.

Reply:
> If you meant emitting warnings: it could go either way. Not strictly necessary, but this would be a very high signal to noise ratio warning.

One issue is that if we emit a warning in `_coerce_pointslike_arg` for an array of shape `(1, 3)` to warn users of the behavior change, it means that even after the user adjusts their downstream code, it will still emit the warning.  I like the approach of adding to the docstring a clear description of the change, so users can figure out what happened.

Reply:
Yeah, I realized that too eventually. It would be useful the first time but useless afterward. Agreed on the doc annotation.

Reply:
I used `..versionchanged:: 0.35.0`, but I'm not 100% confident that this is formatted correctly in the right place.

Reply:
I see you've pushed a corresponding chance since. Looks good to me now, resolving.